src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs

```diff
@@ -351,25 +366,40 @@ private void EventLoop()
         private void RequestEventLoopShutdown()
         {
             //
-            // Write to the pipe, which will wake up the event loop and cause it to exit.
+            // Write to the pipe or eventfd, which will wake up the event loop and cause it to exit.
+            // (Need to write 8 bytes in case we're on a system where eventfd is supported.)
             //
-            byte b = 1;
-            int bytesWritten = Interop.Sys.Write(_shutdownWritePipe, &b, 1);
-            if (bytesWritten != 1)
+            byte[] wakeThreadUp = {0, 0, 0, 0, 0, 0, 0, 1};
+            fixed (byte *b = &wakeThreadUp[0])
```

Comment:
This allocates an 8-byte array on the heap.  That isn't necessary.  You can instead make it:
```C#
byte* wakeThreadUp = stackalloc byte[8] { 0, 0, 0, 0, 0, 0, 0, 1 };
```
and not need the `fixed` at all.

Reply:
Thanks.  Still working on the intuition to understand how C# code is generated/executed.